src/pykeen/nn/emb.py

```diff
@@ -43,6 +87,38 @@ def reset_parameters(self) -> None:
     def post_parameter_update(self):
         """Apply constraints which should not be included in gradients."""
 
+    def get_in_canonical_shape(
+        self,
+        indices: Optional[torch.LongTensor] = None,
+    ) -> torch.FloatTensor:
+        """Get representations in canonical shape.
+
+        :param indices: None, shape: (b,) or (b, n)
+            The indices. If None, return all representations.
+
+        :return: shape: (b?, n?, d)
+            If indices is None, b=1, n=max_id.
+            If indices is 1-dimensional, b=indices.shape[0] and n=1.
+            If indices is 2-dimensional, b, n = indices.shape
+        """
+        x = self(indices=indices)
+        if indices is None:
+            x = x.unsqueeze(dim=0)
+        else:
+            if indices.ndimension() == 1:
+                x = x.unsqueeze(dim=1)
+            elif indices.ndimension() > 2:
+                raise ValueError(
+                    f"Undefined canonical shape for more than 2-dimensional index tensors: {indices.shape}"
+                )
+        return x
+
+    @property
+    def embedding_dim(self) -> int:
+        """Return the "embedding dimension". Kept for backward compatibility."""
+        # TODO: Remove this property and update code to use shape instead
```

Comment:
maybe add a `warnings.warn(..., Deprecated)` to make mean logging

Reply:
cf. 
4f34aa4